### PR TITLE
Improve docs for KSHC

### PR DESCRIPTION
**Pull request by pan3793**
# :mag: Description

Canonicalize the words, and enrich the description for KSCH.

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Test Plan 🧪

Review.

---

# Checklist 📝

- [x] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

**Be nice. Be informative.**

### CODE REVIEW

**Comment by pan3793** [comment]
cc @cxzl25 @cfmcgrady @yikf @zhouyifan279 @FANNG1

**Review by dnskr — CHANGES_REQUESTED** [review]
Left some change suggestions.

File: docs/connector/spark/hive.rst [hunk 1]
```diff
@@ -16,29 +16,29 @@
...
 Dependencies
 ************
 
-The **classpath** of kyuubi spark sql engine with Hive connector supported consists of
+The **classpath** of Kyuubi Spark SQL engine with Hive connector supported consists of
 
-1. kyuubi-spark-connector-hive_2.12-\ |release|\ , the hive connector jar deployed with Kyuubi distributions
+1. kyuubi-spark-sql-engine-\ |release|\ _2.12.jar, the engine jar deployed with Kyuubi distributions
 2. a copy of spark distribution
+3. kyuubi-spark-connector-hive_2.12-\ |release|\ , which can be found in the `Maven Central`_
```
Comment (FANNG1) [inline docs/connector/spark/hive.rst:41]: replace 2.12 with ${scala_version}?

File: docs/connector/spark/hive.rst [hunk 2]
```diff
@@ -106,4 +101,29 @@ Taking ``DROP NAMESPACE`` as a example,
 
    DROP NAMESPACE hive_catalog.ns;
 
-.. _Apache Spark: https://spark.apache.org/
\ No newline at end of file
+Advanced Usages
+***************
+
+Though KSCH is a pure Spark DataSource V2 connector which isn't coupled with Kyuubi deployment, due to the
```
Comment (FANNG1) [inline docs/connector/spark/hive.rst:107]: should we suggest users to set `spark.sql.catalog.hive_catalog.spark.sql.hive.metastore.jars` explicitly to bypass some problems if using `spark-sql`?

File: docs/connector/spark/hive.rst [hunk 3]
```diff
@@ -16,29 +16,29 @@
 `Hive`_
 ==========
```
Comment (FANNG1) [inline docs/connector/spark/hive.rst:17]: should we add supported Spark version? Iceberg, Hudi version for other section

File: docs/connector/spark/hive.rst (hunk 2 above)
Comment (pan3793) [reply docs/connector/spark/hive.rst:107]: No, I would not recommend using `IsolatedClassloader`, for any purpose. Actually, if you read the code of `spark-sql`, you may find that it claims it does not support `IsolatedClassloader`, I'm not sure why such a combination works, maybe another side-effect of a bug.

File: docs/connector/spark/delta_lake.rst [hunk 4]
```diff
@@ -16,35 +16,35 @@
...
 
 .. tip::
    This article assumes that you have mastered the basic knowledge and
    operation of `Delta Lake`_.
-   For the knowledge about delta lake not mentioned in this article,
+   For the knowledge about Delta Lake not mentioned in this article,
    you can obtain it from its `Official Documentation`_.
 
-By using kyuubi, we can run SQL queries towards delta lake which is more
+By using kyuubi, we can run SQL queries towards Delta Lake which is more
```
[the remark this answers is not in the document]
Comment (pan3793) [reply docs/connector/spark/delta_lake.rst:29]: thanks for checking, let me have another around look.

File: docs/connector/spark/hive.rst (hunk 3 above)
Comment (pan3793) [reply docs/connector/spark/hive.rst:17]: Add a brief description for version compatibility.

File: docs/connector/spark/hive.rst [hunk 5]
```diff
@@ -106,4 +105,29 @@ Taking ``DROP NAMESPACE`` as a example,
 
    DROP NAMESPACE hive_catalog.ns;
 
-.. _Apache Spark: https://spark.apache.org/
\ No newline at end of file
+Advanced Usages
+***************
+
+Though KSCH is a pure Spark DataSource V2 connector which isn't coupled with Kyuubi deployment, due to the
```
Comment (DenineLu) [inline docs/connector/spark/hive.rst:111]: It should be KSHC

Comment (pan3793) [reply docs/connector/spark/hive.rst]: My bad, fixed.

**Review by cfmcgrady — APPROVED** [review]
LGTM

**Comment by pan3793** [comment]
Thanks, merged to master/1.9

**Pull request closed without merging**